src/main/java/org/opensearch/ad/model/DataByFeatureId.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.ad.model;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+import java.io.IOException;
+
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.xcontent.ToXContent.Params;
+import org.opensearch.common.xcontent.ToXContentObject;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+
+import com.google.common.base.Objects;
+
+/**
+ * Data and its Id
+ *
+ */
+public class DataByFeatureId implements ToXContentObject, Writeable {
+
+    public static final String FEATURE_ID_FIELD = "feature_id";
+    public static final String DATA_FIELD = "data";
+
+    protected String featureId;
+    protected Double data;
+
+    public DataByFeatureId(String featureId, Double data) {
+        this.featureId = featureId;
+        this.data = data;
+    }
+
+    /*
+     * Used by the subclass that has its own way of initializing data like
+     * reading from StreamInput
+     */
+    protected DataByFeatureId() {}
+
+    public DataByFeatureId(StreamInput input) throws IOException {
+        this.featureId = input.readString();
+        this.data = input.readDouble();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        XContentBuilder xContentBuilder = builder.startObject().field(FEATURE_ID_FIELD, featureId).field(DATA_FIELD, data);
+        return xContentBuilder.endObject();
+    }
+
+    public static DataByFeatureId parse(XContentParser parser) throws IOException {
+        String featureId = null;
+        Double data = null;
+
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+        while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+            String fieldName = parser.currentName();
+            parser.nextToken();
+
+            switch (fieldName) {
+                case FEATURE_ID_FIELD:
+                    featureId = parser.text();
+                    break;
+                case DATA_FIELD:
+                    data = parser.doubleValue();
+                    break;
+                default:
+                    break;
```

Comment:
Should we add `parser.skipChildren();` before break?

Reply:
so skipChildren means if we saw some unknown field, we skip its children?

Reply:
Yes, the unknown field and it's children should be ignored, this is what AnomalyDetector does to not parse unnecessary fields.

Reply:
fixed.